src/machine.rs

```diff
@@ -1009,10 +1010,15 @@ impl OlmMachine {
             for room_room_keys_entry in room_room_keys.entries() {
                 let room_room_keys_entry: Array = room_room_keys_entry?.dyn_into()?;
                 let session_id: JsString = room_room_keys_entry.get(0).dyn_into()?;
-                let key: BackedUpRoomKey =
-                    serde_wasm_bindgen::from_value(room_room_keys_entry.get(1))?;
-
-                keys.entry(room_id.clone()).or_default().insert(session_id.into(), key);
+                serde_wasm_bindgen::from_value::<BackedUpRoomKey>(room_room_keys_entry.get(1))
+                    .map_or_else(
+                        |_| {
+                            failures = failures + 1;
+                        },
+                        |key| {
+                            keys.entry(room_id.clone()).or_default().insert(session_id.into(), key);
+                        },
```

Comment:
personally I'd find a `match` statement easier to read here, but ymmv

Reply:
Indeed, since the error is ignored anyways, even an if/else would read better: `if let Ok(key) = serde_wasm_bindgen::... { keys.entry(...)... } else { failures += 1; }`